File: code-samples/team-messaging/quick-start.php
<?php
// Remember to modify the path ./../ pointing to the location where the RingCentral SDK was installed and the .env file was saved!
require('./../vendor/autoload.php');
$dotenv = Dotenv\Dotenv::createImmutable(__DIR__ . './../');
$dotenv->load();

# Instantiate the SDK and get the platform instance
$rcsdk = new RingCentral\SDK\SDK( $_ENV['RC_APP_CLIENT_ID'],
                                  $_ENV['RC_APP_CLIENT_SECRET'],
                                  $_ENV['RC_SERVER_URL'] );
$platform = $rcsdk->platform();

// Authenticate a user using a personal JWT token
try {
  $platform->login( [ "jwt" => $_ENV['RC_USER_JWT'] ] );
  create_team();
} catch (\RingCentral\SDK\Http\ApiException $e) {
  exit("Unable to authenticate to platform. Check credentials. " . $e->message . PHP_EOL);
}

/*
* Create a new public team in Team Messaging with 3 internal members including the team owner
*/
function create_team(){
  global $platform;
  try {
    $bodyParams = array(
          'public' => true,
          'name' => "PHP Team",
          // Add internal members using their extension ids
          // Get your user extension id by calling the /restapi/v1.0/account/~/extension endpoint!
          'members' => array ( array ( 'id' => "590490017"), array ( 'id' => "595861017" )),
          // You can also add members using their email address, especially for guest members who are not under your account company.
          // 'members' => array(array('email => "member.1@gmail.com"), array('email' => "member.2@gmail.com"), array('id' => "extensionId")),
          'description' => "Let's talk about PHP"
    );
    $endpoint = "/team-messaging/v1/teams";
    $resp = $platform->post($endpoint, $bodyParams);
    print($resp->text());
  }catch (\RingCentral\SDK\Http\ApiException $e) {
    // Getting error messages using PHP native interface
    print 'HTTP Error: ' . $e->getMessage() . PHP_EOL;
    // Another way to get message, but keep in mind, that there could be no response if request has failed completely
    print 'Unable to create a new team. ' . $e->apiResponse->response()->error() . PHP_EOL;
  }
}
?>


<?php
/**********************************************************
 Code snippet section for boostrap testing purpose
**********************************************************/
boostrap_test_function();
function boostrap_test_function(){
/*
  sleep(2);
  print_r ("Test creating compliance export task". PHP_EOL);
  require_once (__DIR__ .'/code-snippets/compliance-export.php');

  sleep(2);
  print_r ("Test reading teams". PHP_EOL);
  require_once (__DIR__ .'/code-snippets/list-teams.php');
*/
  sleep(2);
  print_r ("Test adding team members". PHP_EOL);
  require_once (__DIR__ .'/code-snippets/add-team-members.php');
}
?>

File: code-samples/team-messaging/quick-start.rb
require 'ringcentral'
require 'dotenv'
# Remember to modify the path to where you saved your .env file!
Dotenv.load("./../.env")

#
# Create a new public team in Team Messaging with 2 members including the team owner
#
def create_team()
  begin
    bodyParams = {
      'public': true,
      'name': "Ruby Team",
      # Add internal members using their extension ids
      # Get your user extension id by calling the /restapi/v1.0/account/~/extension endpoint!
      'members': [{ 'id': "590490017"}, { 'id': "595861017"}],
      # You can also add members using their email address, especially for guest members who are not under your account company.
      # 'members': [{'email': "member.1@gmail.com"}, { 'email': "member.2@gmail.com"}, {'id': "[extensionId]"}],
      'description': "Let's talk about Ruby"
    }
    endpoint = "/team-messaging/v1/teams"
    resp = $platform.post(endpoint, payload: bodyParams)
    puts resp.body
  rescue StandardError => e
    puts ("Unable to create a new team. " + e.to_s)
  end
end

# Instantiate the SDK and get the platform instance
$platform = RingCentral.new( ENV['RC_APP_CLIENT_ID'], ENV['RC_APP_CLIENT_IDCLIENT_SECRET'], ENV['RC_SERVER_URL'] )

# Authenticate a user using a personal JWT token
def login()
  begin
    $platform.authorize(jwt: ENV['RC_USER_JWT'])
    create_team()
  rescue StandardError => e
    puts ("Unable to authenticate to platform. Check credentials." + e.to_s)
  end
end

login()
##############################
# End of quick start section
##############################


########
# Code snippet section for boostrap testing purpose
########
def boostrap_test_function()

    # sleep(2)
    # puts "Test creating compliance export task"
    # require_relative './code-snippets/compliance-export'
    # create_compliance_export_task()

    # sleep(2)
    # puts "Test reading teams"
    # require_relative './code-snippets/list-teams'
    # list_teams("")

    sleep(2)
    puts "Test adding new team members"
    require_relative './code-snippets/add-team-members'
    find_team("", "Ruby Team")
end

boostrap_test_function()
